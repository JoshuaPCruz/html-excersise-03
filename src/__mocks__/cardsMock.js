import { cards } from '../manifests/CardList.json';

export default cards;