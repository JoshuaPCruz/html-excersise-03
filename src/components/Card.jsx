import React from 'react';
import '../css/components/Card.css';

const Card = ({image, title, body}) => {
  return (
    <figure className="Card">
      <img src={image} alt="" />
      <figcaption>
      <h4>{title}</h4>
      <p>{body}</p>
      </figcaption>
    </figure>
  );
};

export default Card;