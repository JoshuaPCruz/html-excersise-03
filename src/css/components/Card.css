.Card {
  display: flex;
  flex-flow: column;
  max-width: 285px;
  min-width: 285px;
  height: 550px;
  border: 2px solid var(--gray-primary);
}

.Card figcaption {
  margin: 20px;
}

.Card h4 {
  font-size: 28px;
  padding: 0 12px 16px;
  margin-bottom: 20px;
  border-bottom: 2px solid var(--gray-primary);
  color: #496B97;
}

.Card p {
  font-size: 17px;
  color: var(--gray-primary);
}