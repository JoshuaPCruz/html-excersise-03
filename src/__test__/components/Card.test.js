import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Card from '../../components/Card';
import cardsMock from '../../__mocks__/cardsMock';

describe('<Card />', () => {
  const card = shallow(<Card {...cardsMock[0]} />);
  test('Render del componente Card', () => {
    expect(card.length).toEqual(1);
  });

  test('It received the props', () => {
    expect(card.find('h4').text()).toBe(cardsMock[0].title);
  });
});

describe('Card Snapshot', () => {
  test('Comprobar la UI del componente Card', () => {
    const card = create(<Card {...cardsMock[0]} />);
    expect(card.toJSON()).toMatchSnapshot();
  });
});