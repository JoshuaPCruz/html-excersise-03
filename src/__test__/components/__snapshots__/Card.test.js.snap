// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshot Comprobar la UI del componente Card 1`] = `
<figure
  className="Card"
>
  <img
    alt=""
    src="../src/images/dudes 1.jpg"
  />
  <figcaption>
    <h4>
      Consultor√≠a
    </h4>
    <p>
      Duis autem velm iriure dolor in hendrerit vulputate...
    </p>
  </figcaption>
</figure>
`;
